lib/postgrex/producer.ex

```diff
@@ -29,6 +29,13 @@ defmodule Postgrex.Producer do
     execute on error, otherwise set to `:transaction` (default: `:transaction`);
 
   The pool may support other options.
+
+  ## Examples
+
+      {:ok, pid} = Postgrex.Producer.start_link(pool, "SELECT * FROM posts, [])
+      [{pid, [cancel: :transient]}]
+      |> GenStage.stream()
```

Comment:
Instead of this example, let's make an imaginary consumer subscribe to the producer. That's because this use case is better done with streams.